apis/python/src/tiledbsoma/_experiment.py

```diff
@@ -5,16 +5,40 @@
 
 """Implementation of a SOMA Experiment.
 """
+from typing import Any, Optional
 
-from typing import Any
-
-from somacore import experiment
+import numpy as np
+import pandas as pd
+import tiledb
+from somacore import experiment, query
 
+from . import pytiledbsoma as clib
 from ._collection import Collection, CollectionBase
 from ._dataframe import DataFrame
 from ._measurement import Measurement
 from ._tdb_handles import Wrapper
 from ._tiledb_object import AnyTileDBObject
+from .options import SOMATileDBContext
+from .options._soma_tiledb_context import _validate_soma_tiledb_context
+
+
+def _indexer_map_locations(keys: np.typing.NDArray[np.int64]) -> clib.IntIndexer:
+    if len(np.unique(keys)) != len(keys):
+        raise pd.errors.InvalidIndexError(
+            "Reindexing only valid with uniquely valued Index objects"
+        )
+    context: SOMATileDBContext = _validate_soma_tiledb_context(
+        SOMATileDBContext(tiledb.default_ctx())
+    )
+    compute_concurrency: int = 5
+    if context._tiledb_ctx:
+        compute_concurrency = int(
+            int(context._tiledb_ctx.config()["sm.compute_concurrency_level"]) / 2
+        )
```

Comment:
I think I composed a reply on the last revision of this but GitHub ate it or I forgot to save or something.

This ends up constructing an entirely new TileDB Context, which is one thing we want to avoid. This is something we should get from the context that the Experiment already owns.

Reply:
Fixed